ui/raidboss/data/05-shb/alliance/the_puppets_bunker.js

```diff
@@ -1,10 +1,261 @@
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Dodge Sword Charges',
+      },
+    },
+    {
+      id: 'Puppet Superior Precision Guided Missile',
+      netRegex: NetRegexes.startsUsing({ id: '4CF5', capture: false }),
+      suppressSeconds: 5,
+      alertText: function(data) {
+        // This gets casted by all the planes, so harder to collect.
+        // Not worth saying "on you" or meticulously telling the healer who their tank is.
+        if (data.role === 'tank' || data.role === 'healer') {
+          return {
+            en: 'Tank Buster',
+            de: 'Tank buster',
+            fr: 'Tank buster',
+            ja: 'タンクバスター',
+            cn: '坦克死刑',
+            ko: '탱버',
+          };
+        }
+        return {
+          en: 'Avoid tank cleave',
+          de: 'Tank Cleave ausweichen',
+          fr: 'Évitez le tank cleave',
+          ja: '前方範囲攻撃を避け',
+          ko: '광역 탱버 피하기',
+          cn: '远离顺劈',
+        };
+      },
+    },
+    {
+      id: 'Puppet Superior Sliding Swipe First',
+      netRegex: NetRegexes.startsUsing({ id: ['4FA[CD]', '550[DEF]', '5510'] }),
+      durationSeconds: 6,
+      preRun: function(data) {
+        data.swipe = data.swipe || [];
+        data.swipe.push({
+          '4FAC': Responses.goRight(),
+          '4FAD': Responses.goLeft(),
+          '550D': Responses.goRight(),
+          '550E': Responses.goLeft(),
+          '550F': Responses.goRight(),
+          '5510': Responses.goLeft(),
+        });
+      },
+      response: function(data, matches) {
+        if (data.swipe.length !== 1)
+          return;
+
+        // Call and clear the first swipe so we can not call it a second time below.
+        const swipe = data.swipe[0];
+        data.swipe[0] = null;
+        return swipe(data, matches);
+      },
+    },
+    {
+      id: 'Puppet Superior Sliding Swipe Others',
+      netRegex: NetRegexes.ability({ id: ['4FA[CD]', '550[DEF]', '5510'] }),
+      response: function(data, matches) {
+        if (!data.swipe)
+          return;
+
+        // The first swipe callout has been cleared to null.
+        // Deliberately skip it so that when the first swipe goes off, we call the second.
+        let swipe = data.swipe.pop();
+        if (!swipe)
+          swipe = data.swipe.pop();
+        if (!swipe)
+          return;
+        return swipe(data, matches);
+      },
+    },
+    {
+      id: 'Puppet Heavy Volt Array',
+      netRegex: NetRegexes.startsUsing({ source: '905P-Operated Heavy Artillery Unit', id: '5006', capture: false }),
+      condition: Conditions.caresAboutAOE(),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Puppet Heavy Active Laser Turret',
+      netRegex: NetRegexes.startsUsing({ source: '905P-Operated Heavy Artillery Unit', id: '4FED', capture: false }),
+      // TODO: how to even describe this.  Could have a second callout when lower disppears?
+      alertText: {
+        en: 'Upper Laser Outside -> Lower',
+      },
+    },
```

Comment:
I do think it's better to do two separate calls here, either now or in the next release. This can be a hectic mechanic for those of us who don't play physical ranged. Maybe call the first one as "Outside on high lasers"?

Reply:
What text would you use for these two triggers? I'm failing to come up with anything pithy.

Reply:
I'm similarly unsure. "Outside: On High Lasers" and "Move To Low Lasers" maybe? It could be we have to just accept that the user has to do the thing whether or not we can tell them exactly what it is. Actually, is there even a log entry for "low laser disappears"?